extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/K8sTaskAdapter.java

```diff
@@ -0,0 +1,277 @@
+    long dbbSize = heapSize;
+    if (optionalDbb.isPresent()) {
+      dbbSize = optionalDbb.get();
+    }
+    return (long) ((dbbSize + heapSize) * 1.2);
+
+  }
+
+  protected void setupPorts(Container mainContainer)
+  {
+    mainContainer.getPorts().clear();
+    ContainerPort tcpPort = new ContainerPort();
+    tcpPort.setContainerPort(DruidK8sConstants.PORT);
+    tcpPort.setName("druid-port");
+    tcpPort.setProtocol("TCP");
+    ContainerPort httpsPort = new ContainerPort();
+    httpsPort.setContainerPort(DruidK8sConstants.TLS_PORT);
+    httpsPort.setName("druid-tls-port");
+    httpsPort.setProtocol("TCP");
+    mainContainer.setPorts(Lists.newArrayList(httpsPort, tcpPort));
+  }
+
+  protected void addEnvironmentVariables(Container mainContainer, PeonCommandContext context, String taskContents)
+  {
+    mainContainer.getEnv().addAll(Lists.newArrayList(
+        new EnvVarBuilder()
+            .withName(DruidK8sConstants.TASK_DIR_ENV)
+            .withValue(context.getTaskDir().getAbsolutePath())
+            .build(),
+        new EnvVarBuilder()
+            .withName(DruidK8sConstants.TASK_JSON_ENV)
+            .withValue(taskContents)
+            .build(),
+        new EnvVarBuilder()
+            .withName(DruidK8sConstants.JAVA_OPTS)
+            .withValue(Joiner.on(" ").join(context.getJavaOpts()))
+            .build(),
+        new EnvVarBuilder()
+            .withName(DruidK8sConstants.DRUID_HOST_ENV)
+            .withValueFrom(new EnvVarSourceBuilder().withFieldRef(new ObjectFieldSelector(
+                null,
+                "status.podIP"
+            )).build()).build(),
+        new EnvVarBuilder()
+            .withName(DruidK8sConstants.DRUID_HOSTNAME_ENV)
+            .withValueFrom(new EnvVarSourceBuilder().withFieldRef(new ObjectFieldSelector(
+                null,
+                "metadata.name"
+            )).build()).build()
+    ));
+  }
+
+  protected Container setupMainContainer(
+      PodSpec podSpec,
+      PeonCommandContext context,
+      long containerSize,
+      String taskContents
+  )
+  {
+    // prepend the startup task.json extraction command
+    List<String> mainCommand = Lists.newArrayList("sh", "-c");
+    // update the command
+    List<Container> containers = podSpec.getContainers();
+    Container mainContainer = Iterables.getFirst(containers, null);
+    if (mainContainer == null) {
+      throw new IllegalArgumentException("Must have at least one container");
+    }
+
+    // remove probes
+    mainContainer.setReadinessProbe(null);
+    mainContainer.setLivenessProbe(null);
+
+    setupPorts(mainContainer);
+    addEnvironmentVariables(mainContainer, context, taskContents);
+
+    mainContainer.setCommand(mainCommand);
+    mainContainer.setArgs(Collections.singletonList(Joiner.on(" ").join(context.getComamnd())));
+
+    mainContainer.setName("main");
+    ImmutableMap<String, Quantity> resources = ImmutableMap.of(
+        "cpu",
+        new Quantity("1000", "m"),
```

Comment:
i think this should be a configuration in KubernetesTaskRunnerConfig as well

Reply:
All tasks use a single core, I don't think there is a need to configure this value.  It would most likely lead to issues if you set it at anything but 1 cpu.